testsuites/syncgateway/upgrade/testserver/sg_single_sgw/test_upgrade.py

```diff
@@ -290,11 +290,14 @@ def test_upgrade(params_from_base_test_setup):
         sdk_docs = {doc['_id']: doc for doc in sdk_doc_bodies}
         sdk_client.upsert_multi(sdk_docs)
         sg_client.add_docs(url=sg_admin_url, db=sg_db, number=num_sdk_docs, id_prefix="sgw_after_upgrade", channels=sg_user_channels)
-        time.sleep(30)  # to let the docs import from server to sgw
         replicator.wait_until_replicator_idle(repl2)
-        cbl_doc_ids2 = db.getDocIds(cbl_db2, limit=num_docs + (num_sdk_docs * 4) + 3)
-        count1 = sum('sdk_after_upgrade' in s for s in cbl_doc_ids2)
-        assert count1 == num_sdk_docs, "docs via sdk after upgrade did not replicate to cbl"
+        if upgraded_xattrs_enabled:
+            time.sleep(30)  # to let the docs import from server to sgw
+            cbl_doc_ids2 = db.getDocIds(cbl_db2, limit=num_docs + (num_sdk_docs * 4) + 3)
+            count1 = sum('sdk_after_upgrade' in s for s in cbl_doc_ids2)
+            assert count1 == num_sdk_docs, "docs via sdk after upgrade did not replicate to cbl"
+        else:
+            cbl_doc_ids2 = db.getDocIds(cbl_db2, limit=num_docs + (num_sdk_docs * 3) + 3)
```

Comment:
do the following 2 line belong to the else condition? line 301 and 302? since if condition has already asserted the count?

Reply:
Xattrs are making it very slow.